### PR TITLE
Runtime warnings

### PR DESCRIPTION
**Pull request by compnerd**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Comment by compnerd** [comment]
@swift-ci please test

File: stdlib/public/runtime/Errors.cpp [hunk 1]
```diff
@@ -140,16 +141,16 @@ void swift::dumpStackTraceEntry(unsigned index, void *framePC,
     fprintf(stderr, "%s`%s + %td", libraryName.data(), symbolName.c_str(),
             offset);
   } else {
-    constexpr const char *format = "%-4u %-34s 0x%0.16lx %s + %td\n";
-    fprintf(stderr, format, index, libraryName.data(), symbolAddr,
-            symbolName.c_str(), offset);
+    constexpr const char *format = "%-4u %-34s 0x%0.16tx %s + %td\n";
+    fprintf(stderr, format, index, libraryName.data(),
+            static_cast<uintptr_t>(symbolAddr), symbolName.c_str(), offset);
```
Comment (jrose-apple) [inline stdlib/public/runtime/Errors.cpp]: `symbolAddr` is already a `uintptr_t`.

Comment (compnerd) [reply stdlib/public/runtime/Errors.cpp]: GCC 6 seemed to emit conversion warnings for it even with `%tx`.  I think it was a `void *`.

Comment (jrose-apple) [reply stdlib/public/runtime/Errors.cpp]: `framePC` is a `void *`, but `symbolAddr` isn't. Copy/paste from below, maybe?

Comment (compnerd) [reply stdlib/public/runtime/Errors.cpp]: I'm going to go with, I'm just losing my mind.  Verified that it isn't an issue.  Thanks for catching that!

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
I believe all of @jrose-apple's comments have been address.

**Pull request merged**